### PR TITLE
small systemd patches

### PR DESCRIPTION
**Pull request by etbe**

### CODE REVIEW

File: policy/modules/system/init.if [hunk 1]
```diff
@@ -3263,6 +3263,24 @@ interface(`init_tcp_recvfrom_all_daemons',`
...
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`init_restart_units',`
+	gen_require(`
+		type init_var_run_t;
+	')
+
+	allow $1 init_var_run_t:service { start status stop };
```
Comment (pebenito) [inline policy/modules/system/init.if:3281]: There already are interfaces for transient units (and a new type).

File: policy/modules/system/systemd.if [hunk 2]
```diff
@@ -61,6 +57,7 @@ template(`systemd_role_template',`
 	allow $1_systemd_t $3:process { setsched rlimitinh signal_perms };
 	corecmd_shell_domtrans($1_systemd_t, $3)
 	corecmd_bin_domtrans($1_systemd_t, $3)
+	corecmd_shell_entry_type($1_systemd_t)
```
Comment (pebenito) [inline policy/modules/system/systemd.if:60]: When is this needed?

File: policy/modules/system/systemd.if [hunk 3]
```diff
@@ -153,6 +150,12 @@ template(`systemd_role_template',`
 	allow $3 systemd_user_runtime_t:lnk_file { manage_lnk_file_perms relabel_lnk_file_perms };
 	allow $3 systemd_user_runtime_t:sock_file { manage_sock_file_perms relabel_sock_file_perms };
 
+	# for "machinectl shell"
+	allow $1_systemd_t systemd_machined_t:fd use;
+	allow $3 systemd_machined_t:fd use;
+	allow $3 systemd_machined_t:dbus send_msg;
+	allow systemd_machined_t $3:dbus send_msg;
```
Comment (pebenito) [inline policy/modules/system/systemd.if:157]: This doesn't seem like something we'd want on all users.

**Review by pebenito — CHANGES_REQUESTED** [review]

File: policy/modules/system/init.if (hunk 1 above)
Comment (etbe) [reply policy/modules/system/init.if]: ok

File: policy/modules/system/systemd.if (hunk 2 above)
Comment (etbe) [reply policy/modules/system/systemd.if:60]: Doesn't seem to be needed now so I'll remove it.

File: policy/modules/system/systemd.if (hunk 3 above)
Comment (etbe) [reply policy/modules/system/systemd.if:156]: Why not? systemd-machined requires authentication before doing things.

**Review by pebenito — APPROVED** [review]

**Pull request merged**